Project/app/src/main/java/org/ionproject/android/schedule/ScheduleViewModel.kt

```diff
@@ -0,0 +1,94 @@
+package org.ionproject.android.schedule
+
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.liveData
+import org.ionproject.android.common.model.Lecture
+import org.ionproject.android.common.repositories.CalendarTermRepository
+import org.ionproject.android.common.repositories.ClassesRepository
+import org.ionproject.android.common.repositories.EventsRepository
+import org.ionproject.android.common.repositories.FavoriteRepository
+
+class ScheduleViewModel(
+    private val favoriteRepository: FavoriteRepository,
+    private val calendarTermRepository: CalendarTermRepository,
+    private val classesRepository: ClassesRepository,
+    private val eventsRepository: EventsRepository
+) : ViewModel() {
+
+    /**
+     * Get most recent calendar term
+     * Get all favorites from that calendar term
+     * Get the details of all favorites
+     * Get all lectures from those favorites
+     * Sort all lectures according to the days of week
+     * post to live data to notify observers
+     */
+    private val lecturesLiveData = liveData {
+        val calendarTerm = calendarTermRepository.getMostRecentCalendarTerm()
+        val favorites = favoriteRepository.suspendGetFavoritesFromTerm(calendarTerm)
+        val classSections = favorites.map {
+            classesRepository.getClassSection(it.toClassSummary())
+        }
+        val lectures = classSections.flatMap {
+            eventsRepository.getLectures(it?.calendarURI)
+        }
+        emit(sortLecturesByDay(lectures))
+    }
+
+    fun observerLecturesLiveData(
+        lifecycleOwner: LifecycleOwner,
+        onUpdate: (Array<MutableList<Lecture>>) -> Unit
+    ) {
+        lecturesLiveData.observe(lifecycleOwner, Observer { onUpdate(it) })
+    }
+
+    private fun sortLecturesByDay(lectures: List<Lecture>): Array<MutableList<Lecture>> {
+        val lecturesByDayOfWeek2 = arrayOf<MutableList<Lecture>>(
```

Comment:
Ugly! You can do better, I bet. ;)
What about something like this: arrayOf(size) { mutableListOf() }
It does not exist, but you could create it... ;)